Sources/Task/TaskResult.swift

```diff
@@ -7,14 +7,23 @@
 //
 
 /// An enum for returning and propogating recoverable errors.
-public enum TaskResult<Value> {
+#if swift(>=3.1)
+@available(*, deprecated, message: "Use Task.Result instead.")
+public typealias TaskResult<SuccessValue> = Task<SuccessValue>.Result
+#else
+public enum TaskResult<SuccessValue> {
     /// Contains the success value
-    case success(Value)
+    case success(SuccessValue)
     /// Contains the error value
     case failure(Error)
 }
+#endif
+
+#if swift(>=3.1)
+extension Task.Result: Either {
+
+    public typealias Value = SuccessValue
```

Comment:
Add an `@available(*, deprecated, renamed: "SuccessValue")` to both of these. Might get some more warnings from that.